Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -100,9 +112,92 @@ class HistoryPanel: SiteTableViewController, HomePanel {
         }
     }
 
+    func updateNumberOfSyncedDevices() {
+        var numberOfDevices = 0
+        syncedTabsHeaderCell.descriptionLabel?.hidden = true
+        syncedTabsHeaderCell.descriptionLabel?.removeFromSuperview()
+        syncedTabsHeaderCell.descriptionLabel = nil
+
+        self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
+            if let clientAndTabs = result.successValue {
+                let nonEmptyClientAndTabs = clientAndTabs.filter { $0.tabs.count > 0 }
+                numberOfDevices = nonEmptyClientAndTabs.count
+            }
+
+            if numberOfDevices > 0 {
+                self.syncedTabsHeaderCell.descriptionLabel = UILabel()
+                self.syncedTabsHeaderCell.descriptionLabel?.font = DynamicFontHelper.defaultHelper.DeviceFontSmallHistoryPanel
+                self.syncedTabsHeaderCell.descriptionLabel?.textColor = UIColor.grayColor()
+                self.syncedTabsHeaderCell.descriptionLabel?.text = String(numberOfDevices) + " " + Strings.SyncedTabsTableViewCellDescription
+            }
```

Comment:
Nit: You can clean up this logic a bit by using guard and a conditional instead:

```swift
guard let clientAndTabs = result.successValue,
      let numberOfDevices = clientAndTabs.filter { $0.tabs.count > 0 } where numberOfDevices > 0 else {
    self.syncedTabsHeaderCell.descriptionLabel.hidden = true
    return
}

self.syncedTabsHeaderCell.descriptionLabel.text = String(numberOfDevices) + " " + Strings.SyncedTabsTableViewCellDescription
self.syncedTabsHeaderCell.descriptionLabel.hidden = false
```